lib/output/writer/blob_storage.go

```diff
@@ -0,0 +1,178 @@
+	StorageAccessKey string `json:"storage_access_key" yaml:"storage_access_key"`
+	Container        string `json:"container" yaml:"container"`
+	Path             string `json:"path" yaml:"path"`
+	BlobType         string `json:"blob_type" yaml:"blob_type"`
+	Timeout          string `json:"timeout" yaml:"timeout"`
+	MaxInFlight      int    `json:"max_in_flight" yaml:"max_in_flight"`
+}
+
+// NewAzureBlobStorageConfig creates a new Config with default values.
+func NewAzureBlobStorageConfig() AzureBlobStorageConfig {
+	return AzureBlobStorageConfig{
+		StorageAccount:   "",
+		StorageAccessKey: "",
+		Container:        "",
+		Path:             `${!count("files")}-${!timestamp_unix_nano()}.txt`,
+		BlobType:         "BLOCK",
+		Timeout:          "5s",
+		MaxInFlight:      1,
+	}
+}
+
+//------------------------------------------------------------------------------
+
+// AzureBlobStorage is a benthos writer. Type implementation that writes messages to an
+// Azure Blob Storage storage account.
+type AzureBlobStorage struct {
+	conf      AzureBlobStorageConfig
+	container field.Expression
+	path      field.Expression
+	blobType  field.Expression
+	timeout   time.Duration
+	log       log.Modular
+	stats     metrics.Type
+}
+
+// NewAzureBlobStorage creates a new Amazon S3 bucket writer.Type.
+func NewAzureBlobStorage(
+	conf AzureBlobStorageConfig,
+	log log.Modular,
+	stats metrics.Type,
+) (*AzureBlobStorage, error) {
+	var timeout time.Duration
+	if tout := conf.Timeout; len(tout) > 0 {
+		var err error
+		if timeout, err = time.ParseDuration(tout); err != nil {
+			return nil, fmt.Errorf("failed to parse timeout period string: %v", err)
+		}
+	}
+	a := &AzureBlobStorage{
+		conf:    conf,
+		log:     log,
+		stats:   stats,
+		timeout: timeout,
+	}
+	var err error
+	if a.container, err = field.New(conf.Container); err != nil {
+		return nil, fmt.Errorf("failed to parse container expression: %v", err)
+	}
+	if a.path, err = field.New(conf.Path); err != nil {
+		return nil, fmt.Errorf("failed to parse path expression: %v", err)
+	}
+	if a.blobType, err = field.New(conf.BlobType); err != nil {
+		return nil, fmt.Errorf("failed to parse blob type expression: %v", err)
+	}
+	return a, nil
+}
+
+// ConnectWithContext attempts to establish a connection to the target Blob Storage Account.
+func (a *AzureBlobStorage) ConnectWithContext(ctx context.Context) error {
+	return a.Connect()
+}
+
+// Connect attempts to establish a connection to the target Blob Storage Account.
+func (a *AzureBlobStorage) Connect() error {
+	return nil
+}
+
+// Write attempts to write message contents to a target Azure Blob Storage container as files.
+func (a *AzureBlobStorage) Write(msg types.Message) error {
+	return a.WriteWithContext(context.Background(), msg)
+}
+
+func (a *AzureBlobStorage) getContainer(name string) (*azblob.ContainerURL, error) {
+	accountName, accountKey := a.conf.StorageAccount, a.conf.StorageAccessKey
+	if len(accountName) == 0 || len(accountKey) == 0 {
```

Comment:
This check ought to be done within `NewAzureBlobStorage` so that users can see the error during start up.

Reply:
You're right. I'm checking now within `NewAzureBlobStorage`